malariagen_data/ag3.py

```diff
@@ -341,3 +354,73 @@ def _results_cache_add_analysis_params(self, params):
         super()._results_cache_add_analysis_params(params)
         # override parent class to add AIM analysis
         params["aim_analysis"] = self._aim_analysis
+
+    @check_types
+    @doc(
+        summary="Load tag SNPs for a given inversion in Ag.",
+    )
+    def load_inversion_tags(self, inversion: inversion_param) -> pd.DataFrame:
+        # needs to be modified depending on where we are hosting
+        import importlib.resources
+        from . import resources
+
+        with importlib.resources.path(resources, "karyotype_tag_snps.csv") as path:
+            df_tag_snps = pd.read_csv(path, sep=",")
+        return df_tag_snps.query(f"inversion == '{inversion}'").reset_index()
+
+    @check_types
+    @doc(
+        summary="Infer karyotype from tag SNPs for a given inversion in Ag.",
+    )
+    def karyotype(
+        self,
+        inversion: inversion_param,
+        sample_sets: Optional[base_params.sample_sets] = None,
+        sample_query: Optional[base_params.sample_query] = None,
+    ) -> pd.DataFrame:
+
+        # load tag snp data
+        df_tagsnps = self.load_inversion_tags(inversion=inversion)
+        inversion_pos = df_tagsnps["position"]
+        inversion_alts = df_tagsnps["alt_allele"]
+        contig = inversion[0:2]
+
+        # get snp calls for inversion region
+        start, end = np.min(inversion_pos), np.max(inversion_pos)
+        region = f"{contig}:{start}-{end}"
+
+        ds_snps = self.snp_calls(
+            region=region, sample_sets=sample_sets, sample_query=sample_query
+        )
+        geno = allel.GenotypeDaskArray(ds_snps["call_genotype"].data)
+        pos = allel.SortedIndex(ds_snps["variant_position"].values)
+        samples = ds_snps["sample_id"].values
+        alts = ds_snps["variant_allele"].values.astype(str)
+
+        # subset to position of inversion tags
+        mask = pos.locate_intersection(inversion_pos)[0]
+        alts = alts[mask]
+        geno = geno.compress(mask, axis=0).compute()
+
+        with self._spinner("Inferring karyotype from tag SNPs"):
+            gn_alt = _karyotype_tags_n_alt(
+                gt=geno, alts=alts, inversion_alts=inversion_alts
+            )
+            is_called = geno.is_called()
+
+            # calculate mean genotype for each sample whilst masking missing calls
+            av_gts = np.mean(np.ma.MaskedArray(gn_alt, mask=~is_called), axis=0)
+            total_sites = np.sum(is_called, axis=0)
+
+            df = pd.DataFrame(
+                {
+                    "sample_id": samples,
+                    "inversion": inversion,
+                    "karyotype_mean": av_gts,
+                    # round the genotypes then convert to int
+                    "karyotype": av_gts.round().astype(int),
```

Comment:
```suggestion
                    f"karyotype_{inversion}_mean": av_gts,
                    # round the genotypes then convert to int
                    f"karyotype_{inversion}": av_gts.round().astype(int),
```

Just a thought, would make it easier if someone then wants to join together results for multiple inversions into a single dataframe, save them having to rename the "karyotype" columns in each.